drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -921,50 +974,53 @@
       them to stop functioning properly. Operators deploying IOAM with this
       Integrity Protection Method MUST ensure that such overload situations are
       avoided. This could for example be achieved by applying IOAM only to a
-      subset of the entire traffic, keeping in mind that only that subset would
-      be integrity protected.</t>
+      subset of the entire traffic, keeping in mind that only that IOAM subset
+      would be integrity protected. When enabled, the integrity protection MUST
+      be applied to the entire IOAM set, not a subset.</t>
+
+      <t>A compromised IOAM transit node could reinitialize both the Nonce and
+      ICV fields in the Integrity Protection header, in order to pretend to be
+      an encapsulating node and fool the Validator. Therefore, the compromised
+      node could modify IOAM-Data-Fields and cause the integrity validation
+      to succeed. To avoid such a situation, the Validator MUST know all
+      encapsulating nodes currently allowed, and so for each IOAM Namespace-ID,
+      thanks to, e.g., a list of encapsulating node identifiers. Implementation
```

Comment:
How about:

"[...] To avoid such a situation, the Validator MUST be aware of all IOAM nodes configured as encapsulating nodes, and so for each IOAM Namespace-ID respectively. This could be achieved by sharing with the Validator, e.g., a list of encapsulating node identifiers for each IOAM Namespace-ID. Implementation [...]"

Reply:
Still a bit confused... Does the first sentence mean:

_To avoid such a situation, the Validator MUST know all IOAM encapsulating nodes and the IOAM Namespace-IDs configured on each of the IOAM encapsulating nodes?_ 

If so, then I start to follow.

Reply:
> Still a bit confused... Does the first sentence mean:
> 
> _To avoid such a situation, the Validator MUST know all IOAM encapsulating nodes and the IOAM Namespace-IDs configured on each of the IOAM encapsulating nodes?_
> 
> If so, then I start to follow.

Indeed. I'll use your text, it is clearer :-)